### PR TITLE
fix: `# of Activities` of active members change based on the time frame

### PR DESCRIPTION
**Pull request by yeganathan18**
# Changes proposed ✍️

### <samp>🤖 Generated by Copilot at b9369bf</samp>

Replaced `list` method with `listActive` method in `actions.js` to filter and sort active members more efficiently and accurately. Part of a dashboard module refactor.
​

### <samp>🤖 Generated by Copilot at b9369bf</samp>

> _`listActive` method_
> _filters and sorts by activity_
> _dashboard refactored_

### <samp>🤖 Generated by Copilot at b9369bf</samp>

* Replace `list` method of `MemberService` with `listActive` method to simplify query logic and measure member activity more accurately ([link](https://github.com/CrowdDotDev/crowd.dev/pull/792/files?diff=unified&w=0#diff-3229e811e36fcc52729fe2443f9ca14ae324f62b35ac2d6aaf15daf705b6bd1dL160-R172))

## Checklist ✅
- [ ] Label appropriately with `Feature`, `Improvement`, or `Bug`.
- [ ] Add screehshots to the PR description for relevant FE changes
- [ ] New backend functionality has been unit-tested.
- [ ] API documentation has been updated (if necessary) (see [docs on API documentation](https://docs.crowd.dev/docs/updating-api-documentation)).
- [ ] [Quality standards](https://github.com/CrowdDotDev/crowd-github-test-public/blob/main/CONTRIBUTING.md#quality-standards) are met.

### CODE REVIEW

**Comment by yeganathan18** [comment]
This PR hopefully should fix the issue #755 🍀

**Comment by yeganathan18** [comment]
Here's how it looks after the fix, works as intended now!

https://user-images.githubusercontent.com/63534555/233610342-25c35e5f-2309-418a-8668-dbd2774091f6.mov

**Review by joanagmaia — CHANGES_REQUESTED** [review]
Thanks for the contribution! 😄 
Could you update `orderBy` from `activeDaysCount_DESC` to `activityCount_DESC`?
That way member with higher activityCount (which is the value we are displaying) can be sorted properly

**Comment by yeganathan18** [comment]
@joanagmaia Oops, sorry I missed that. Now updated the PR with changes!

**Review by joanagmaia — APPROVED** [review]

File: frontend/src/modules/dashboard/store/actions.js [hunk 1]
```diff
@@ -157,41 +157,19 @@ export default {
...
-            : []),
-        ],
-      },
-      'activityCount_DESC',
-      5,
-      0,
-      false,
-    )
+    return MemberService.listActive({
+      platform: platform !== 'all' ? platform : [],
```
Comment (joanagmaia) [inline frontend/src/modules/dashboard/store/actions.js:161]: Just noticed another issue while testing this locally.
When you change the period all good 👌 but if you change the platform filter to a specific platform (you need to have more than 1 integration for it to appear), the list of members is not updated.
The issue is with the way you are passing `platform` to the `listActive` method.

If you notice there `platform` must be an array of objects. So for this to work you would need to update this to something like this:
```
platform: platform !== 'all' ? [{ value: platform }] : [],
```
Screenshot 2023-04-21 at 17 34 18

**Review by joanagmaia — CHANGES_REQUESTED** [review]

File: frontend/src/modules/dashboard/store/actions.js (hunk 1 above)
Comment (yeganathan18) [reply frontend/src/modules/dashboard/store/actions.js]: Pushed the changes, thanks for letting me know.

I'm marking the conversation as resolved and also will ensure no bugs are introduced next time with the code 🙂

**Review by joanagmaia — APPROVED** [review]
LGTM 🚀

**Pull request merged**